tiny_cnn/layers/layer.h

```diff
@@ -238,55 +240,79 @@ class layer_base {
         return true;
     }
 
+    virtual void set_worker_count(cnn_size_t worker_count) {
+        if (worker_count == 0) {
+            throw nn_error("worker_count cannot be zero!");
+        }
+
+        if (worker_count != worker_storage_.size()) {
+            worker_storage_.resize(worker_count);
+
+            // resize the vectors of each worker
+            set_size(in_size_, out_size_, W_.size(), b_.size());
+        }
+    }
+
 protected:
     cnn_size_t in_size_;
     cnn_size_t out_size_;
     bool parallelize_;
 
     layer_base* next_;
     layer_base* prev_;
-    vec_t a_[CNN_TASK_SIZE];          // w * x
```

Comment:
This is the beef of the pull request. These should not be fixed-size, unless n_threads given to train() is somehow limited. (Currently it isn't, so we may overflow these arrays – in particular when not using OpenMP, in which case `CNN_TASK_SIZE` is only 8.)